data/modules/Scout/ScanManager.lua

```diff
@@ -425,12 +438,19 @@ function ScanManager:CanScanBeActivated(id)
 
 	if scan.complete then return false end
 
-	local frameBody = self.ship.frameBody
-	if not frameBody then return false end
+	local body = self:GetScanBodyByType(scan)
+	if not body then return false end
```

Comment:
If the player leaves the frame of influence of the body they are scanning, they are no longer in orbit of that body. They are instead orbiting the parent body and will slowly begin drifting away from the scanned body unless they enter the frame of influence again.

The problem at hand is that the frame of influence of single planets or moons without any children is far smaller than it should be. This is due to Space.cpp#761 using `4 * bodyRadius` instead of calculating the Hill sphere of the body for its correct gravitational frame of influence.

The equation for calculating the Hill radius (in pseudocode) is:

```c
semiMajorAxis * (1.0 - eccentricity) * pow(mass / (3.0 * parent.mass), 1.0 / 3.0)
```

Doing a quick evaluation, using the Hill radius for frame-of-influence should ensure that most if not all bodies are fully scannable while remaining inside their frame of influence, as the Hill radius for small bodies can be 10x or larger than the current frame of influence calculation.

Reply:
Do it actually needed since surface scan's are already performed on the low altitude and for orbtial scan's, body borrows from scan's body path and the distance is checked by current resolution:

`function ScanManager:GetScanBodyByType(scan)`
```
return scan.orbital and scan.bodyPath:GetSystemBody().body or self.ship.frameBody
```

`function ScanManager:CanScanBeActivated(id)`
```
if scan.orbital then
        local altitude, resolution = self:GetBodyState(body, scan)
	if(resolution > (scan.minResolution * 1.10)) then
		return false
	end
end
```
?
But feel free to correct me if I misunderstood.

Reply:
Yes, this change is needed. I can push it as a new commit to this branch if you're interested; but essentially it modifies how Pioneer's orbital physics simulation works to make it both more physically correct and also more fun for the player.

Reply:
Yes, do it.